packages/app/src/systems/CRX/background/services/BackgroundService.ts

```diff
@@ -307,13 +309,31 @@ export class BackgroundService {
     return currentAccount?.address;
   }
 
-  async network() {
+  async network(): Promise<Network> {
     const selectedNetwork = await NetworkService.getSelectedNetwork();
+
+    if (!selectedNetwork) return { chainId: 0, url: '' };
+
+    const isValidChainId = !Number.isNaN(Number(selectedNetwork.id));
+
     return {
-      url: selectedNetwork?.url,
+      chainId: isValidChainId ? Number(selectedNetwork.id) : 0,
+      url: selectedNetwork.url,
     };
   }
 
+  async networks(): Promise<Network[]> {
+    const networks = await NetworkService.getNetworks();
+    return networks.map((network) => {
+      const isValidChainId = !Number.isNaN(Number(network.id));
+
+      return {
+        chainId: isValidChainId ? Number(network.id) : 0,
```

Comment:
what is the use case to have a nan chainId? we should never have it

Reply:
it was just because i was confusing the `uuid` as the `chainId`
now that we'll add one more column just for that, i'll remove these NaN checks